src/js_lexer.zig

```diff
@@ -328,39 +342,33 @@ fn NewLexer_(
 
                             // legacy octal literals
                             '0'...'7' => {
-                                const octal_start = (iter.i + width2) - 2;
                                 if (comptime is_json) {
-                                    lexer.end = start + iter.i - width2;
-                                    try lexer.syntaxError();
+                                    lexer.end = start + iter.i + 1;
```

Comment:
I think these `lexer.end` updates do nothing, but I am not sure. Should I remove them?